File: CS/MVVMDemo.Wpf/Modules/AsyncCommands/AsyncPOCOCommandsViewModel.cs
using DevExpress.Mvvm.POCO;
using System;
using System.Threading;
using System.Threading.Tasks;

namespace MVVMDemo.AsyncCommands {
    public class AsyncPOCOCommandsViewModel {
        public virtual int Progress { get; set; }

        public Task Calculate() {
            return Task.Factory.StartNew(CaclulateCore);
        }
        void CaclulateCore() {
            for(int i = 0; i <= 100; i++) {
                if(this.GetAsyncCommand(x => x.Calculate()).IsCancellationRequested) {
                    Progress = 0;
                    return;
                }
                Progress = i;
                Thread.Sleep(TimeSpan.FromMilliseconds(20));
            }
        }
    }
}

File: VB/MVVMDemo.Wpf/Modules/AsyncCommands/AsyncPOCOCommandsViewModel.vb
Imports DevExpress.Mvvm.POCO
Imports System
Imports System.Threading
Imports System.Threading.Tasks

Namespace MVVMDemo.AsyncCommands
    Public Class AsyncPOCOCommandsViewModel
        Public Overridable Property Progress() As Integer

        Public Function Calculate() As Task
            Return Task.Factory.StartNew(AddressOf CaclulateCore)
        End Function
        Private Sub CaclulateCore()
            For i As Integer = 0 To 100
                If Me.GetAsyncCommand(Function(x) x.Calculate()).IsCancellationRequested Then
                    Progress = 0
                    Return
                End If
                Progress = i
                Thread.Sleep(TimeSpan.FromMilliseconds(20))
            Next i
        End Sub
    End Class
End Namespace
